src/open_inwoner/openzaak/tests/test_case_detail_redirects.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib.messages import get_messages
+from django.test import TestCase, override_settings
+from django.urls import reverse
+
+from open_inwoner.openzaak.models import ZGWApiGroupConfig
+from open_inwoner.openzaak.tests.factories import ZGWApiGroupConfigFactory
+
+
+@override_settings(
+    ROOT_URLCONF="open_inwoner.cms.tests.urls",
+)
+class LegacyCaseDetailUrlRedirectTest(TestCase):
+    def setUp(self):
+        self.url_patterns = [
+            "case_detail",
+        ]
+
+    def test_legacy_url_redirects_to_only_api_group_prefix(self):
+        first = ZGWApiGroupConfigFactory()
+
+        object_id = "test_object_id"
+        legacy_handler_url = reverse(
+            "cases:legacy_case_detail",
+            kwargs={"object_id": object_id},
+        )
+        target_url = reverse(
+            "cases:case_detail",
+            kwargs={"api_group_id": first.id, "object_id": object_id},
+        )
+
+        response = self.client.get(legacy_handler_url)
+        self.assertEqual(ZGWApiGroupConfig.objects.count(), 1)
```

Comment:
Is this necessary? You created one, so there exists one. The only way this assertion would add something if it was possible that additional `ZGWApiGroupConfig` objects get created during the action (i.e. `self.client.get()`), which is not the case.